src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Prerendering.cs

```diff
@@ -146,6 +148,7 @@ internal async ValueTask<PrerenderedComponentHtmlContent> RenderEndpointComponen
         {
             var component = BeginRenderingComponent(rootComponentType, parameters);
             var result = new PrerenderedComponentHtmlContent(Dispatcher, component);
+            stopAddingTasks = httpContext.Response.StatusCode == StatusCodes.Status404NotFound && _hasStatusCodePage;
```

Comment:
What would happen if we just remove `httpContext.Response.StatusCode == StatusCodes.Status404NotFound` and use only `_hasStatusCodePage`?

Reply:
It works but before doing so, I would like to clear the situation about re-executing what was not intended to re-execute [see test](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1007939&view=ms.vss-test-web.build-test-results-tab&runId=26929680&resultId=100003&paneView=debug).

Reply:
It's passing on CI, I have to try harder to clean the test app.